pkg/uprobetracer/ldcache_parser.go

```diff
@@ -0,0 +1,199 @@
+	err := binary.Read(buffer, binary.NativeEndian, obj)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func readStringFromBytes(data []byte, startPos uint32) string {
+	res := ""
+	for i := startPos; i < uint32(len(data)); i++ {
+		if data[i] == 0 {
+			return res
+		}
+		res += string(data[i])
+	}
+	return ""
+}
+
+func readCacheFormat1(data []byte) []ldEntry {
+	var ldEntries []ldEntry
+
+	ldCache := ldCache1{}
+	if uint32(len(data)) <= ldCache1Size {
+		return nil
+	}
+	err := readFromBytes(&ldCache, data[:ldCache1Size])
+	if err != nil {
+		return nil
+	}
+	ldEntriesOffset := ldCache1Size
+	ldStringsOffset := ldCache1Size + ldCache1EntrySize*ldCache.EntryCount
+	for i := uint32(0); i < ldCache.EntryCount; i++ {
+		entryOffset := ldEntriesOffset + i*ldCache1EntrySize
+		entry := ldCache1Entry{}
+		if uint32(len(data)) <= entryOffset+ldCache1EntrySize {
+			return nil
+		}
+		err := readFromBytes(&entry, data[entryOffset:entryOffset+ldCache1EntrySize])
+		if err != nil {
+			return nil
+		}
+		keyOffset := ldStringsOffset + entry.Key
+		valueOffset := ldStringsOffset + entry.Value
+		key := readStringFromBytes(data, keyOffset)
+		value := readStringFromBytes(data, valueOffset)
+		ldEntries = append(ldEntries, ldEntry{key, value})
+	}
+	return ldEntries
+}
+
+func readCacheFormat2(data []byte) []ldEntry {
+	var ldEntries []ldEntry
+
+	if !bytes.Equal([]byte(cache2Header), data[:cache2HeaderLen]) {
+		return nil
+	}
+	ldCache := ldCache2{}
+	if uint32(len(data)) <= ldCache2Size {
+		return nil
+	}
+	err := readFromBytes(&ldCache, data[:ldCache2Size])
+	if err != nil {
+		return nil
+	}
+	ldEntriesOffset := ldCache2Size
+	for i := uint32(0); i < ldCache.EntryCount; i++ {
+		entryOffset := ldEntriesOffset + i*ldCache2EntrySize
+		entry := ldCache2Entry{}
+		if uint32(len(data)) <= entryOffset+ldCache2EntrySize {
+			return nil
+		}
+		err := readFromBytes(&entry, data[entryOffset:entryOffset+ldCache2EntrySize])
+		if err != nil {
+			return nil
+		}
+		keyOffset := entry.Key
+		valueOffset := entry.Value
+		key := readStringFromBytes(data, keyOffset)
+		value := readStringFromBytes(data, valueOffset)
+		ldEntries = append(ldEntries, ldEntry{key, value})
+	}
+	return ldEntries
+}
+
+// simulate the loader's behaviour, find library path in containers' `/etc/ld.so.cache`.
+// see https://github.com/bminor/glibc/blob/master/elf/cache.c#L292, the `print_cache` func
+// see https://github.com/iovisor/bcc/blob/master/src/cc/bcc_proc.c#L508, the `bcc_procutils_which_so` func
+func parseLdCache(ldCachePath string, libraryName string) []string {
+	fileInfo, err := os.Stat(ldCachePath)
+	if err != nil {
+		return nil
+	}
+	if fileInfo.Mode()&os.ModeNamedPipe != 0 {
+		return nil
+	}
+	ldCacheFile, err := os.ReadFile(ldCachePath)
+	if err != nil {
+		return nil
+	}
+	ldCacheFileSize := uint32(len(ldCacheFile))
+
+	var ldEntries []ldEntry
+	var filteredLibraries []string
+
+	if bytes.Equal([]byte(cache1Header), ldCacheFile[:cache1HeaderLen]) {
```

Comment:
```suggestion
	if bytes.Equal([]byte(cache1Header), ldCacheFile[:len(cache1Header)]) {
```

Since `cache1Header` is a constant, `len(cache1Header)` is also a constant.

Ditto in similar code pattern.